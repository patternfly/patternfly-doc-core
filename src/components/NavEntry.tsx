import { NavItem } from '@patternfly/react-core'
import { kebabCase } from '../utils/case'

export interface TextContentEntry {
  id: string
  data: {
    id: string
    section: string
    tab?: string
    sortValue?: number
  }
}

interface NavEntryProps {
  entry: TextContentEntry
  isActive: boolean
}

export const NavEntry = ({ entry, isActive }: NavEntryProps) => {
  const { id: contentLocationId } = entry
  const { id: entryTitle, section } = entry.data

  const kebabTitle = kebabCase(entryTitle)
  const kebabSection = kebabCase(section)

  return (
    <NavItem
      itemId={contentLocationId}
      to={`/${kebabSection}/${kebabTitle}`}
      isActive={isActive}
      id={`nav-entry-${contentLocationId}`}
    >
      {entryTitle}
    </NavItem>
  )
}
