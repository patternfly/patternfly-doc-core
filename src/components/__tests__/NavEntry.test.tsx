import { render, screen } from '@testing-library/react'
import { NavEntry, TextContentEntry } from '../NavEntry'

const mockEntry: TextContentEntry = {
  id: 'entry1',
  data: { id: 'Entry 1 Title', section: 'section1' },
}

describe('NavEntry', () => {
  it('renders without crashing', () => {
    render(<NavEntry entry={mockEntry} isActive={false} />)
    expect(screen.getByText('Entry 1 Title')).toBeInTheDocument()
  })

  it('renders the correct link with kebab-cased title', () => {
    render(<NavEntry entry={mockEntry} isActive={false} />)
    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      '/section1/entry-1-title',
    )
  })

  it('properly transforms PatternFly text in URLs', () => {
    const patternflyEntry: TextContentEntry = {
      id: 'pf1',
      data: { id: 'PatternFly Components', section: 'PatternFly' },
    }
    render(<NavEntry entry={patternflyEntry} isActive={false} />)
    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      '/patternfly/patternfly-components',
    )
  })

  it('preserves PatternFly text in display', () => {
    const patternflyEntry: TextContentEntry = {
      id: 'pf1',
      data: { id: 'PatternFly Components', section: 'PatternFly' },
    }
    render(<NavEntry entry={patternflyEntry} isActive={false} />)
    expect(screen.getByText('PatternFly Components')).toBeInTheDocument()
  })

  it('marks the entry as active if isActive is true', () => {
    render(<NavEntry entry={mockEntry} isActive={true} />)
    expect(screen.getByRole('link')).toHaveClass('pf-m-current')
  })

  it('does not mark the entry as active if isActive is false', () => {
    render(<NavEntry entry={mockEntry} isActive={false} />)
    expect(screen.getByRole('link')).not.toHaveClass('pf-m-current')
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <NavEntry entry={mockEntry} isActive={false} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
