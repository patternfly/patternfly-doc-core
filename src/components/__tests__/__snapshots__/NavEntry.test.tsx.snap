// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavEntry matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="pf-v6-c-nav__item"
    data-ouia-component-id="OUIA-Generated-NavItem-7"
    data-ouia-component-type="PF6/NavItem"
    data-ouia-safe="true"
  >
    <a
      class="pf-v6-c-nav__link"
      href="/section1/entry-1-title"
      id="nav-entry-entry1"
    >
      <span
        class="pf-v6-c-nav__link-text"
      >
        Entry 1 Title
      </span>
    </a>
  </li>
</DocumentFragment>
`;
